test/model/autocomplete_test.dart

```diff
@@ -1,14 +1,166 @@
+import 'package:fake_async/fake_async.dart';
 import 'package:test/scaffolding.dart';
 import 'package:zulip/api/model/model.dart';
 import 'package:zulip/model/autocomplete.dart';
+import 'package:zulip/model/narrow.dart';
 
 import '../example_data.dart' as eg;
+import 'test_store.dart';
 
 void main() {
+  test('MentionAutocompleteView misc', () async {
+    const narrow = AllMessagesNarrow();
+    final store = eg.store()
+      ..addUsers([eg.selfUser, eg.otherUser, eg.thirdUser]);
+    final view = MentionAutocompleteView.init(store: store, narrow: narrow);
+
+    bool done = false;
+    view.addListener(() { done = true; });
+    view.query = MentionAutocompleteQuery('Third');
+    await Future(() {});
+    check(done).isTrue();
+    check(view.results).single
+      .isA<UserMentionAutocompleteResult>()
+      .has((r) => r.userId, 'userId')
+      .equals(eg.thirdUser.userId);
+  });
+
+  test('MentionAutocompleteView not starve timers', () {
+    fakeAsync((binding) {
+      const narrow = AllMessagesNarrow();
+      final store = eg.store()
+        ..addUsers([eg.selfUser, eg.otherUser, eg.thirdUser]);
+      final view = MentionAutocompleteView.init(store: store, narrow: narrow);
+
+      bool searchDone = false;
+      view.addListener(() {
+        searchDone = true;
+      });
+
+      // Schedule a timer event with zero delay.
+      // This stands in for a user interaction, or frame rendering timer,
+      // placing an urgent task on the event queue.
+      bool timerDone = false;
+      Timer(const Duration(), () {
+        timerDone = true;
+        // The timer should go first, before the search does its work.
+        check(searchDone).isFalse();
+      });
+
+      view.query = MentionAutocompleteQuery('Third');
+      check(timerDone).isFalse();
+      check(searchDone).isFalse();
+
+      binding.elapse(const Duration(seconds: 1));
+
+      check(timerDone).isTrue();
+      check(searchDone).isTrue();
+      check(view.results).single
+        .isA<UserMentionAutocompleteResult>()
+        .has((r) => r.userId, 'userId')
+        .equals(eg.thirdUser.userId);
+    });
+  });
+
+  test('MentionAutocompleteView yield between batches of 1000', () async {
+    const narrow = AllMessagesNarrow();
+    final store = eg.store();
+    for (int i = 0; i < 3000; i++) {
+      store.addUser(eg.user(userId: i, email: 'user$i@example.com', fullName: 'User $i'));
+    }
+    final view = MentionAutocompleteView.init(store: store, narrow: narrow);
+
+    bool done = false;
+    view.addListener(() { done = true; });
+    view.query = MentionAutocompleteQuery('User 2999');
+
+    await Future(() {});
+    check(done).isFalse();
+    await Future(() {});
+    check(done).isFalse();
+    await Future(() {});
+    check(done).isFalse();
+    await Future(() {});
+    check(done).isTrue();
+    check(view.results).single
+      .isA<UserMentionAutocompleteResult>()
+      .has((r) => r.userId, 'userId').equals(2999);
```

Comment:
should pull these `.has` calls out into a little extension, similar to https://github.com/zulip/zulip-flutter/pull/94#discussion_r1204676900

Here, as a bonus that will dedupe them.